frontend/src/components/admin/question-creation/question-elements/modals/fraction/AddFractionModal.tsx

```diff
@@ -1,20 +1,27 @@
-import React, { useContext } from "react";
+import React, { useContext, useState } from "react";
+import { useDisclosure } from "@chakra-ui/react";
 import { v4 as uuidv4 } from "uuid";
 
 import QuestionEditorContext from "../../../../../../contexts/QuestionEditorContext";
 import type { FractionMetadata } from "../../../../../../types/QuestionMetadataTypes";
+import type { FractionType } from "../../../../../../types/QuestionTypes";
 import { QuestionElementType } from "../../../../../../types/QuestionTypes";
 
+import ChooseFractionTypeModal from "./ChooseFractionTypeModal";
 import FractionModal from "./FractionModal";
 
 const AddFractionModal = (): React.ReactElement => {
-  const { showAddFractionModal, setShowAddFractionModal } = useContext(
-    QuestionEditorContext,
-  );
-  const closeModal = () => {
-    setShowAddFractionModal(false);
-  };
+  const [fractionType, setFractionType] = useState<FractionType>("regular");
+  const {
+    onOpen: onFractionInputModalOpen,
+    isOpen: isFractionInputModalOpen,
+    onClose: onFractionInputModalClose,
+  } = useDisclosure();
 
+  const {
+    showAddFractionModal: isChooseFractionTypeModalOpen,
```

Comment:
decided to use alias here for clarity, but didn't rename `showAddFractionModal` entirely to decouple the fraction modal flow from the parent components which don't need to know abt its implementation